meta-mender-core/classes/mender-helpers.bbclass

```diff
@@ -350,19 +357,49 @@ def get_extra_parts_by_id(d, id):
 
 def get_extra_parts_labels(d):
     labels = []
-    parts = get_extra_parts(d)
-    if parts:
-        for part in parts:
-            s = part.find("--label")
-            if s > 0:
-                e = part.find("--", s + 2)
-                labels.append(part[s + len("--label="):e])
+    ids = get_extra_parts_flags(d)
+    if ids:
+        for id in ids:
+            labels.append(get_extra_parts_label(d, id))
     return ' '.join(labels)
 
+def get_extra_parts_label(d, id = None):
+    part = get_extra_parts_by_id(d, id)
+    if part:
+        s = part.find("--label")
+        if s > 0:
+            e = part.find("--", s + 2)
+            return part[s + len("--label="):e]
```

Comment:
Upon trying this I realized that this doesn't work if the `--label` argument is the last on the line. The indexed searches are also a bit hard to follow, so I suggest changing it to something that works in both cases and is easier to read, like this:

```diff
diff --git a/meta-mender-core/classes/mender-helpers.bbclass b/meta-mender-core/classes/mender-helpers.bbclass
index 3100fc55..3f75f2f3 100644
--- a/meta-mender-core/classes/mender-helpers.bbclass
+++ b/meta-mender-core/classes/mender-helpers.bbclass
@@ -366,10 +366,9 @@ def get_extra_parts_labels(d):
 def get_extra_parts_label(d, id = None):
     part = get_extra_parts_by_id(d, id)
     if part:
-        s = part.find("--label")
-        if s > 0:
-            e = part.find("--", s + 2)
-            return part[s + len("--label="):e]
+        import re
+        match = re.search(r'--label=(\S+)', part)
+        return match.group(1) if match else ""
     return ""
 
 def get_extra_parts_wks(d):
```

The overhead of using a regex here is negligible anyway.

Reply:
OK fine I'll squash and update